Data/Text/ICU/Spoof.hsc

```diff
@@ -0,0 +1,273 @@
+  toBitMask SkeletonSingleScript = #const USPOOF_SINGLE_SCRIPT_CONFUSABLE
+  toBitMask SkeletonAnyCase = #const USPOOF_ANY_CASE
+
+type USkeletonTypeOverride = Int32
+
+makeSpoofCheckResult :: USpoofCheck -> SpoofCheckResult
+makeSpoofCheckResult c =
+  case spoofChecks of
+    [] -> CheckOK
+    _ ->
+      case restrictionLevel of
+        Nothing -> CheckFailed spoofChecks
+        Just l -> CheckFailedWithRestrictionLevel spoofChecks l
+  where spoofChecks = fromBitMask $ fromIntegral $ c .&. #const USPOOF_ALL_CHECKS
+        restrictionValue = c .&. #const USPOOF_RESTRICTION_LEVEL_MASK
+        restrictionLevel = highestValueInBitMask $ fromIntegral $ restrictionValue
+
+-- | Open a spoof checker for checking Unicode strings for lookalike security issues.
+open :: IO MSpoof
+open = wrap =<< handleError uspoof_open
+
+-- | Open a spoof checker previously serialized to bytes using 'serialize'.
+-- The ForeignPtr in the ByteString will remain referenced for the
+-- lifetime of the returned object. The memory it points to must not
+-- be modified until the returned object is closed.
+openFromSerialized :: ByteString -> IO MSpoof
+openFromSerialized b =
+  case toForeignPtr b of
+    (ptr, off, len) -> withForeignPtr ptr $ \p ->
+      wrapWithSerialized ptr =<< handleError (uspoof_openFromSerialized (p `plusPtr` off) (fromIntegral len) nullPtr)
+
+-- | Open a spoof checker given the contents of the "confusables.txt" and "confusablesWholeScript.txt"
+-- files as described in Unicode UAX #39.
+openFromSource :: ByteString -> ByteString -> IO MSpoof
+openFromSource confusables confusablesWholeScript =
+  unsafeUseAsCStringLen confusables $ \(cptr, clen) ->
+    unsafeUseAsCStringLen confusablesWholeScript $ \(wptr, wlen) ->
+      wrap =<< handleError (uspoof_openFromSource cptr (fromIntegral clen) wptr (fromIntegral wlen) nullPtr nullPtr)
+
+-- | Get the checks performed by a spoof checker.
+getChecks :: MSpoof -> IO [SpoofCheck]
+getChecks s = do
+  withSpoof s $ \sptr ->
+    (fromBitMask . fromIntegral . (.&.) #{const USPOOF_ALL_CHECKS}) <$> handleError (uspoof_getChecks sptr)
+
+-- | Configure the checks performed by a spoof checker.
+setChecks :: MSpoof -> [SpoofCheck] -> IO ()
+setChecks s c = do
+  withSpoof s $ \sptr ->
+    handleError $ uspoof_setChecks sptr . fromIntegral $ toBitMask c
+
+-- | Get the restriction level of a spoof checker.
+getRestrictionLevel :: MSpoof -> IO (Maybe RestrictionLevel)
+getRestrictionLevel s = do
+  withSpoof s $ \sptr ->
+    (highestValueInBitMask . fromIntegral) <$> uspoof_getRestrictionLevel sptr
+
+-- | Configure the restriction level of a spoof checker.
+setRestrictionLevel :: MSpoof -> RestrictionLevel -> IO ()
+setRestrictionLevel s l = do
+  withSpoof s $ \sptr ->
+    uspoof_setRestrictionLevel sptr . fromIntegral $ toBitMask l
+
+-- | Get the list of locale names allowed to be used with a spoof checker.
+-- (We don't use LocaleName since the root and default locales have no meaning here.)
```

Comment:
If you surround type or function names with single quotes, they'll get hyperlinked and rendered properly in the docs.

It's a good idea to run `cabal haddock` over the package and look at the rendered docs (it'll print out the location of the `index.html` file for you, and also tell you how many of the exports from a module are actually documented).